static/src/javascripts/projects/common/modules/commercial/creatives/hosted-thrasher.js

```diff
@@ -0,0 +1,41 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/template',
+    'common/modules/commercial/hosted-video',
+    'common/modules/commercial/creatives/add-tracking-pixel',
+    'text!common/views/commercial/creatives/hosted-thrasher.html'
+], function (
+    fastdom,
+    template,
+    hostedVideo,
+    addTrackingPixel,
+    hostedVideoStr
+) {
+    var hostedVideoTpl;
+
+    var HostedThrasher = function ($adSlot, params) {
+        this.$adSlot = $adSlot;
+        this.params = params;
+    };
+
+    HostedThrasher.prototype.create = function () {
+        if (!hostedVideoTpl) {
+            hostedVideoTpl = template(hostedVideoStr);
+        }
+
+        var domPromise = fastdom.write(function () {
+            this.$adSlot.append(hostedVideoTpl({ data: this.params}));
+        }.bind(this));
```

Comment:
I never know how I feel about this public-all-the-things strategy for these components. I mean, I know it's the idiom for _this specific part of the codebase_, and I guess they never get exposed to any foreign code, but... would it be nicer if we just passed these values straight into static closure functions? We could keep all this stuff truly private (so no-one working on this code for the first time need worry about access control), and we wouldn't need to mess about binding `this` everywhere.

I'm just thinking aloud here.